objects/src/builder/mod.rs

```diff
@@ -0,0 +1,274 @@
+use crate::{
+    mock::assembler, Account, AccountCode, AccountError, AccountId, AccountStorage, AccountType,
+    AccountVault, Asset, AssetError, Digest, Felt, FungibleAsset, MerkleStore, NonFungibleAsset,
+    NonFungibleAssetDetails, StorageItem, String, ToString, Vec, ZERO,
+};
+use assembly::ast::ModuleAst;
+use rand::{distributions::Standard, Rng};
+
+pub struct AccountStorageBuider {
+    items: Vec<StorageItem>,
+}
+
+/// Builder for an `AccountStorage`, the builder can be configured and used multipled times.
+impl AccountStorageBuider {
+    pub fn new() -> Self {
+        Self { items: vec![] }
+    }
+
+    pub fn add_item(&mut self, item: StorageItem) -> &mut Self {
+        self.items.push(item);
+        self
+    }
+
+    pub fn build(&self) -> AccountStorage {
+        AccountStorage::new(self.items.clone(), MerkleStore::new()).unwrap()
+    }
+}
+
+/// Builder for an `AccountId`, the builder can be configured and used multipled times.
+pub struct AccountIdBuilder<T: Rng> {
+    account_type: AccountType,
+    on_chain: bool,
+    code_root: Digest,
+    storage_root: Digest,
+    rng: T,
+}
+
+impl<T: Rng> AccountIdBuilder<T> {
+    pub fn new(rng: T) -> Self {
+        Self {
+            account_type: AccountType::RegularAccountUpdatableCode,
+            on_chain: false,
+            code_root: Digest::default(),
+            storage_root: Digest::default(),
+            rng,
+        }
+    }
+
+    pub fn account_type(&mut self, account_type: AccountType) -> &mut Self {
+        self.account_type = account_type;
+        self
+    }
+
+    pub fn on_chain(&mut self, on_chain: bool) -> &mut Self {
+        self.on_chain = on_chain;
+        self
+    }
+
+    pub fn code_root(&mut self, code_root: Digest) -> &mut Self {
+        self.code_root = code_root;
+        self
+    }
+
+    pub fn storage_root(&mut self, storage_root: Digest) -> &mut Self {
+        self.storage_root = storage_root;
+        self
+    }
+
+    pub fn build(&mut self) -> Result<AccountId, AccountError> {
```

Comment:
In cases that we do care about the account seed (new accounts) then we should return it to the client. I'm not sure if we should split the interface to introduce a second method `build_with_seed(..) -> Result<(AccountId, Word), AccountError>` or if we should add an argument to the build method `build(.., with_seed: bool) -> Result<(AccountId, Option<Word>), AccountError>`.  What do you think?

Reply:
Agreed, but I'm not sure what is the best approach here. I was trying not to nest these constructors, but we just changed that for the `Account`. Which means we would need to cascade `build_with_seed` there too.

So my idea is to add these once the need comes (probably when the data is serialized to a file, since we will most likely need that data at that point). How does that sound?

Reply:
To work with a new account using the transaction kernel the client must provide the account seed via the advice provider.  If a seed is not provided for a new account then the transaction kernel will fail.  As such, by discarding the seed we can not work with new accounts that are created using the builder.  

This is also somewhat related to the point I made above regarding `AccountIdBuilder`, we are spending time clock time grinding a seed which is discarded - in this case we may as well just grind the `AccountId` directly which will take far less time.